lib/dns_mock/response/message.rb

```diff
@@ -24,10 +28,15 @@ def as_binary_string
       private
 
       attr_reader :dns_answer, :dns_message
+      attr_writer :lookup
 
       def compose_answer
         dns_message.each_question do |hostname, record_type|
-          dns_message.answer.push(*dns_answer.build(hostname, record_type))
+          self.lookup = DnsMock::Response::Message::Lookup.new(
+            question: [hostname, record_type],
+            answer: dns_answer.build(hostname, record_type)
+          )
```

Comment:
This causes `lookup` to be overwritten with each loop iteration. I'm not familiar enough with the inner-workings of DNS. Is there only ever one query per "message"?

Reply:
As I see this array includes only one element. Could you run something to get more then one?